job-service-db/src/main/resources/procedures/processFailedDependentJobs.sql

```diff
@@ -36,13 +36,13 @@ BEGIN
         WITH RECURSIVE all_job_dependencies AS (
             SELECT partition_id, dependent_job_id, job_id
             FROM job_dependency
+            WHERE partition_id = in_partition_id AND job_id = in_job_id
             UNION ALL
             SELECT adj.partition_id, adj.dependent_job_id, jd.job_id
             FROM all_job_dependencies adj
```

Comment:
Does this definitely work?  I see it's going back to `all_job_dependencies` which is now differnt.

Reply:
It should, as previously it was building the table for every job querying that where as now it just builds the table for this job. 
It seemed to on the one call I tried. I don't see any IT for it which is slightly concerning though.

Reply:
Make sure to test it with multiple levels of dependencies.

Reply:
I've tested this on a job with 3 levels of dependents and the full hierarchy was correctly returned.